crates/core/src/surfnet/locker.rs

```diff
@@ -1752,6 +1753,55 @@ impl SurfnetSvmLocker {
 }
 
 impl SurfnetSvmLocker {
+    pub fn get_first_local_slot(&self) -> Slot {
+        self.with_svm_reader(|svm_reader| {
+            svm_reader
+                .blocks
+                .keys()
+                .min()
+                .copied()
+                .expect("Should have blocks")
+        })
+    }
+
+    pub async fn get_block(
+        &self,
+        remote_ctx: &Option<SurfnetRemoteClient>,
+        slot: &Slot,
+        config: &RpcBlockConfig,
+    ) -> SurfpoolContextualizedResult<Option<UiConfirmedBlock>> {
+        let first_local_slot = self.get_first_local_slot();
+
+        let result = match first_local_slot > *slot {
+            true => match remote_ctx {
+                Some(remote_client) => remote_client.get_block(slot, *config).await?,
+                None => return Err(SurfpoolError::slot_too_old(*slot)),
+            },
+            false => match self.get_block_local(slot, config).inner {
+                Some(block) => block,
+                None => return Err(SurfpoolError::slot_too_old(*slot)),
```

Comment:
If we don't have a block locally, that does not mean the slot provided to the user is too old. It means they're requesting a slot that doesn't exist yet. This shouldn't return an error, this should just return `None`